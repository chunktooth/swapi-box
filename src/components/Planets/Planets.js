import React from 'react';
import Card from '../Card/Card';

const Planets = ({data, setFavorites}) => {
	const displayPlanets = data.map((planet, index) => {
		return <Card
			key={index} 
			data={planet} 
			setFavorites={setFavorites}
		/>
	});

	return (
		<article>
			{displayPlanets}
		</article>
	)
}

export default Planets;