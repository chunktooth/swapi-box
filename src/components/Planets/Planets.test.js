import React from 'react';
import Planets from './Planets';
import {shallow} from 'enzyme';

describe('Planets', () => {
	
	const mockFunction = jest.fn();
	const mockData = [{
		name: 'Earth',
		terrain: 'Water',
		population: '0',
		climate: 'Hot',
		residents: 'Lucy'
	}];

	const wrapper = shallow(<Planets
			data={mockData} 
			setFavorites={mockFunction} />)

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
	
});