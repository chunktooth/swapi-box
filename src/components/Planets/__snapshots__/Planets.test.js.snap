// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    data={
        Array [
            Object {
              "climate": "Hot",
              "name": "Earth",
              "population": "0",
              "residents": "Lucy",
              "terrain": "Water",
            },
          ]
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          data={
                    Object {
                              "climate": "Hot",
                              "name": "Earth",
                              "population": "0",
                              "residents": "Lucy",
                              "terrain": "Water",
                            }
          }
          setFavorites={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "climate": "Hot",
            "name": "Earth",
            "population": "0",
            "residents": "Lucy",
            "terrain": "Water",
          },
          "setFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            data={
                        Object {
                                    "climate": "Hot",
                                    "name": "Earth",
                                    "population": "0",
                                    "residents": "Lucy",
                                    "terrain": "Water",
                                  }
            }
            setFavorites={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "climate": "Hot",
              "name": "Earth",
              "population": "0",
              "residents": "Lucy",
              "terrain": "Water",
            },
            "setFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
